balancer-js/src/modules/pools/pool-types/concerns/composableStable/join.concern.integration.spec.ts

```diff
@@ -1,209 +1,165 @@
 
-  before(async function () {
-    this.timeout(20000);
-    const poolObj = pools_16350000.find(
-      ({ id }) =>
-        id ==
-        '0xa13a9247ea42d743238089903570127dda72fe4400000000000000000000035d'
-    ) as unknown as Pool;
-    const initialBalances = poolObj.tokens.map((token) =>
-      parseFixed('100000', token.decimals).toString()
+    const testPool = new TestPoolHelper(
+      testPoolId,
+      network,
+      rpcUrl,
+      blockNumber
     );
-    const slots = [0, 0, 0, 0];
+
+    // Gets initial pool info from Subgraph
+    pool = await testPool.getPool();
+
+    // Setup forked network, set initial token balances and allowances
     await forkSetup(
       signer,
-      poolObj.tokensList,
-      slots,
-      initialBalances,
+      pool.tokensList,
+      Array(pool.tokensList.length).fill(0),
+      Array(pool.tokensList.length).fill(parseFixed('100000', 18).toString()),
       jsonRpcUrl as string,
-      16350000 // holds the same state as the static repository
+      blockNumber // holds the same state as the static repository
     );
-    signerAddress = await signer.getAddress();
-    const bptIndex = poolObj.tokensList.indexOf(poolObj.address);
-    //REMOVING BPT
-    tokensWithoutBpt = [...poolObj.tokensList];
-    tokensWithoutBpt.splice(bptIndex, 1);
-    pool = Pools.wrap(poolObj, networkConfig);
+
+    // Updatate pool info with onchain state from fork block no
+    pool = await testPool.getPool();
   });
-  let joinPoolAttr: JoinPoolAttributes;
-  context('join transaction - join with encoded data', () => {
-    const amountsIn = [
-      parseFixed('100', 18).toString(),
-      parseFixed('101', 18).toString(),
-      parseFixed('102', 18).toString(),
-    ];
-    const slippage = '100';
-
-    before(async function () {
-      this.timeout(20000);
-      [bptBalanceBefore, ...tokensBalanceBefore] = await getBalances(
-        [pool.address, ...tokensWithoutBpt],
+
+  it('should join using encoded data - all tokens have value', async () => {
+    const tokensIn = removeItem(pool.tokensList, pool.bptIndex);
+    const amountsIn = tokensIn.map((_, i) =>
+      parseFixed((i * 100).toString(), 18).toString()
+    );
+    const slippage = BigNumber.from('6');
+    const { to, data, minBPTOut, expectedBPTOut } = pool.buildJoin(
+      signerAddress,
+      tokensIn,
+      amountsIn,
+      slippage.toString()
+    );
+
+    const { transactionReceipt, balanceDeltas } =
+      await sendTransactionGetBalances(
+        pool.tokensList,
         signer,
-        signerAddress
-      );
-      joinPoolAttr = pool.buildJoin(
         signerAddress,
-        tokensWithoutBpt,
-        amountsIn,
-        slippage
+        to,
+        data
       );
 
-      const tx = {
-        to: joinPoolAttr.to,
-        data: joinPoolAttr.data,
-        gasLimit: 3000000,
-      };
+    expect(transactionReceipt.status).to.eq(1);
+    expect(BigInt(expectedBPTOut) > 0).to.be.true;
```

Comment:
I feel like this test is not required, since the following one already covers this case

Reply:
I think it might be useful. If for some reason there is a bug where we encode incorrectly and input tokens are sent but we get no BPT (and expected is 0) then this catches that?